TestAllCurrentObjectsSome test is affected by system git config

**To Reproduce**

1. git-lfs `master` branch
2. `git config --global lfs.storage /path/to/somewhere`
3. `make test`

**Expected behavior**

All tests pass

**Actual behavior**

```
--- FAIL: TestAllCurrentObjectsNone (0.01s)
    lfs_test.go:25: Found: {Oid:d7e099738e67f285241298b4ded39e6db48cc2008fd1960215f3c0403eae3427 Size:12}
    lfs_test.go:25: Found: {Oid:e9d38a101d120a86571af61375b92bc5d219d4e74d205031a021867b0c8ecdb7 Size:30}
    lfs_test.go:25: Found: {Oid:f3c853135aa21c3481c82e917208f0fae5f91eda38032e01dbff95b9c7404c17 Size:25}
    lfs_test.go:25: Found: {Oid:12c2deece9bdb2b80c8098728d68eb1e7bd2ee237ac26c4b5c0c945916cc1b15 Size:20}
    lfs_test.go:25: Found: {Oid:333979845b376fdf6afa0391e9156e686f9b6b2f9ca4dcf9a731e470955f1a36 Size:30}
    lfs_test.go:25: Found: {Oid:c3990ba4bc01a7c0e1d659f8ad4bc1886383246cf85514dfa39508dbce6f6bd3 Size:30}
    lfs_test.go:25: Found: {Oid:c3fb7da348eabe72c12eba33319de9fb80fa97a77788b7da3df1545894351e28 Size:22}
    lfs_test.go:25: Found: {Oid:38aee14d2371d80cb29b10257c5b84992bdb9773a4f0493d4d5f144fddd3c9e7 Size:20}
    lfs_test.go:25: Found: {Oid:35644ac11dc7ba779d277ab5183a313331fd31044d169268641a30a46d01f392 Size:30}
    lfs_test.go:25: Found: {Oid:2487e4f06b0810116fdefb808bd2fc2531c837b4a85ca146642d439fecbfa1f0 Size:30}
    lfs_test.go:25: Found: {Oid:647fc88fa01959a5b455eb4e21a3b3ec729cd467c51198c5144cbf55554bfefa Size:30}
    lfs_test.go:25: Found: {Oid:149a2d01270f8406f1951575c6717b1e2430d529c73a014e74e6599408c27db9 Size:20}
    lfs_test.go:25: Found: {Oid:91c7661e0ab6bcfe6bffa0fa02998b3467d8814e43ac0558adb3ebf86b490b3c Size:32}
    lfs_test.go:25: Found: {Oid:5793be60b7b923be93638041e91ec89d2cb4d6c087634f7ed3d4ec7041731d5a Size:30}
    lfs_test.go:25: Found: {Oid:bff3b771ec7a13b34533022e84895d3c45e1b99157c14a935a5054089c9857e1 Size:40}
    lfs_test.go:25: Found: {Oid:73701626ab91b268bb6830d8d1ca414b0e476786ecb4e2f3f9693c3e4f4ab605 Size:30}
    lfs_test.go:25: Found: {Oid:735ad4fbf949394106c091a78e47fd25d46f03ef29e17d6e7879d8e5a54d6a76 Size:20}
    lfs_test.go:25: Found: {Oid:f859b587469627f138d534b20db381d9f415e0eb138c90914e0f2da33bb2d870 Size:36}
    lfs_test.go:25: Found: {Oid:9b9906f5ae3adb383e5353dd2fa814bf10c2fdc3a8e607e77e0436ec45573ea8 Size:20}
    lfs_test.go:25: Found: {Oid:4b1e147ef485626268a62f22cba6fd5d741671bbb4bdaf7bf00d36dead9dfcda Size:30}
    lfs_test.go:25: Found: {Oid:b4faa9f0b604b349783682706a279715988c2f6447918290f46ebd6e990b2b46 Size:25}
    lfs_test.go:25: Found: {Oid:51b4b333385f4080bf92a1886d0f23f2a82ea0aae0132265bf27b996e056d10c Size:22}
    lfs_test.go:25: Found: {Oid:f57e428d80e5b6b6359b3d9b5b111e4f31af6f3643a6ab2621a0d3301a3e728e Size:30}
    lfs_test.go:25: Found: {Oid:eec2d39c5bc9f7e88c8f85a9d8cb505b82a5bc1c72526d5126a26a8c2fb5bdcb Size:30}
    lfs_test.go:25: Found: {Oid:6c79525a3249c63a06496e266c8b8212cc1f0e260bcf9bd4cad6dc4a71d1bd34 Size:20}
    lfs_test.go:25: Found: {Oid:85cce12a9ef7f0dd0f31cf966101ed3049a17d664fb4ba449f31c0513514fe65 Size:20}
    lfs_test.go:25: Found: {Oid:217b9edf66ab04650bbb438e0b8689fbea56de54a7089a24efba4ec2d0ade974 Size:30}
    lfs_test.go:25: Found: {Oid:65f577efd0e59f25a1ee0a3229261b189d4e14d461a324c671dd3182525a4f35 Size:20}
    lfs_test.go:25: Found: {Oid:f2c2add61f326fa2b1c0420daa5664d423bf82d0fe517b6700879801be0e6b42 Size:30}
    lfs_test.go:25: Found: {Oid:5def0a8e313247dd8c9cf49f1c3eacdeaf86199382494a162459f5aaa396b3c7 Size:40}
    lfs_test.go:25: Found: {Oid:47181e9e688af5a47a932a3713b864c0b9caa08db164556b351f66cc61a02c79 Size:6}
    lfs_test.go:25: Found: {Oid:45851d31df30fb6ada677d95c74c6642cd7b29c9d38066f4bd312409ed05ac9f Size:30}
    lfs_test.go:25: Found: {Oid:4752354602887adb93547d4f1c70cc74b29ba801f126663f836cf0996956fc6b Size:30}
    lfs_test.go:25: Found: {Oid:f781253591c98cdca2e3069f9414579f900205ea3ee435841a0fc6a12c5f165c Size:20}
    lfs_test.go:25: Found: {Oid:2c34f06d7b5af2b406b48bc365888ac1179a176b3b95011afa5919522088f84b Size:21}
    lfs_test.go:25: Found: {Oid:e174da7488e577a5e1fed5bc083f07c927df5e061e5a4c50254898abf1c3c81c Size:20}
    lfs_test.go:25: Found: {Oid:4189151928077250a9817cfbb6811c4194ff06f99d1a42fbd71ab6bda7e21a42 Size:30}
    lfs_test.go:25: Found: {Oid:20e108182c7ec9e6f868cd136e3b51165d2f831d7123174d4bcb2e13448eeb2e Size:25}
    lfs_test.go:25: Found: {Oid:49bbca3e1981b304b6ff5ca71aea2448b505ba50853740553bf887126b0c2654 Size:20}
    lfs_test.go:25: Found: {Oid:9d82a50c56f74a863163213fb2ddd980946ead26d03f1d68500ce1ea724e7c01 Size:31}
    lfs_test.go:25: Found: {Oid:0c304670315211c4e8e6bb953ac09d21bd2a4a275720e950aeea7b27c562e6a8 Size:30}
    lfs_test.go:25: Found: {Oid:081c39282c60731969144980d527570cdf62219e9c64dcfcbba2983eef00a80a Size:30}
    lfs_test.go:25: Found: {Oid:9e366e58df4301d5ceeb4ebee8fd6d4402305ad7b9d6b549de40f3b4de8cf3a6 Size:30}
    lfs_test.go:25: Found: {Oid:231b2ac26933b6ede876ca39be395b0e71600b632c73ad238c17bc521c5b7bd7 Size:20}
    lfs_test.go:25: Found: {Oid:3ac94b1d7c499747864451c13b8abc50a3a196ae54a061654a8bcc0b179327e8 Size:40}
    lfs_test.go:25: Found: {Oid:61f2a9e519d731891da52aab4a84dcc044af59b3b165c9287d397df0e8012007 Size:42}
    lfs_test.go:25: Found: {Oid:f96fdeff073a3145d72b481d532091e3149f0cbea96ce5f8defc2cdce42f46c2 Size:32}
    lfs_test.go:25: Found: {Oid:fda61fdf636be00c1a4883da69315baa452c04f3a111c1378d536c245a0247cb Size:30}
    lfs_test.go:25: Found: {Oid:899b2b5dc70763f362aeb09374d654a937707b930ad21d3531ab6f29181dd508 Size:20}
    lfs_test.go:25: Found: {Oid:974d50fe68c7a0b9958e1ad5247a59b3d311c39adabd1781b53adb9d629dc200 Size:25}
    lfs_test.go:25: Found: {Oid:9668c5a232c8f11ba601af1bf1b4bdc28795bd53c1113138ca75d8ec850f18f2 Size:30}
    lfs_test.go:25: Found: {Oid:ef87221779ec2e218b3a6ef4e1dec9f743b71530b2e2fc561ccdb469195ee800 Size:25}
    lfs_test.go:25: Found: {Oid:0083af118d18a63c6bb552f21d0c4ee78741f988ecd319d3cd06cb6c85a68a63 Size:10}
    lfs_test.go:25: Found: {Oid:373e0b9ea8863fbfea18f30b9cb50f993b5f76fe1be0cca25b1d0a49693bb911 Size:30}
    lfs_test.go:25: Found: {Oid:bd6174d17b30f8dd7e0f19f0553091d33267773fa85bef7b71d8a64d0cb20efd Size:20}
    lfs_test.go:25: Found: {Oid:599712a0858056905f65f47379a656ac40e2429cb87d6ca653986632aa208059 Size:32}
    lfs_test.go:25: Found: {Oid:dcd40b25a27daef8ee0a92aef055376e72c20e227d383b0e124a79b76be13af7 Size:16}
    lfs_test.go:25: Found: {Oid:cccd4b047ed4a6fa5619edfa66d827862e9d637651e28c9aa78131752fda4f3e Size:30}
    lfs_test.go:25: Found: {Oid:826efb18ae20e99a3916053d483629d4a8db495d0e2af8d00cd66f90be588c24 Size:20}
    lfs_test.go:29: Should be no objects
--- FAIL: TestAllCurrentObjectsSome (0.07s)
    lfs_test.go:70: 
                Error Trace:    lfs_test.go:70
                Error:          Not equal: 
                                expected: []*lfs.Pointer{(*lfs.Pointer)(0xc0000ec410), (*lfs.Pointer)(0xc0003320f0), (*lfs.Pointer)(0xc000332140), (*lfs.Pointer)(0xc0000ec280), (*lfs.Pointer)(0xc000332280), (*lfs.Pointer)(0xc0000ec370), (*lfs.Pointer)(0xc000332000), (*lfs.Pointer)(0xc0000ec320), (*lfs.Pointer)(0xc0003320a0), (*lfs.Pointer)(0xc0000ec230), (*lfs.Pointer)(0xc0000ec4b0), (*lfs.Pointer)(0xc000332230), (*lfs.Pointer)(0xc0000ec1e0), (*lfs.Pointer)(0xc0003322d0), (*lfs.Pointer)(0xc000332050), (*lfs.Pointer)(0xc0000ec190), (*lfs.Pointer)(0xc0000ec460), (*lfs.Pointer)(0xc000332190), (*lfs.Pointer)(0xc000332320), (*lfs.Pointer)(0xc0003321e0)}
                                actual  : []*lfs.Pointer{(*lfs.Pointer)(0xc0000ec550), (*lfs.Pointer)(0xc0005ba0a0), (*lfs.Pointer)(0xc0004642d0), (*lfs.Pointer)(0xc0004640a0), (*lfs.Pointer)(0xc000332460), (*lfs.Pointer)(0xc000464190), (*lfs.Pointer)(0xc0003325a0), (*lfs.Pointer)(0xc000464370), (*lfs.Pointer)(0xc0005ba000), (*lfs.Pointer)(0xc0004641e0), (*lfs.Pointer)(0xc000332410), (*lfs.Pointer)(0xc0003324b0), (*lfs.Pointer)(0xc0004be320), (*lfs.Pointer)(0xc0004be050), (*lfs.Pointer)(0xc000464460), (*lfs.Pointer)(0xc000464280), (*lfs.Pointer)(0xc0000ec050), (*lfs.Pointer)(0xc0004640f0), (*lfs.Pointer)(0xc000464140), (*lfs.Pointer)(0xc000464230), (*lfs.Pointer)(0xc0004be0f0), (*lfs.Pointer)(0xc0004be190), (*lfs.Pointer)(0xc0005a00f0), (*lfs.Pointer)(0xc0004be370), (*lfs.Pointer)(0xc0000ec0a0), (*lfs.Pointer)(0xc000464410), (*lfs.Pointer)(0xc0005ba050), (*lfs.Pointer)(0xc0000ec000), (*lfs.Pointer)(0xc000332640), (*lfs.Pointer)(0xc0005a0050), (*lfs.Pointer)(0xc0005a00a0), (*lfs.Pointer)(0xc000332730), (*lfs.Pointer)(0xc0003325f0), (*lfs.Pointer)(0xc0000ec140), (*lfs.Pointer)(0xc000332500), (*lfs.Pointer)(0xc0000ec500), (*lfs.Pointer)(0xc0004644b0), (*lfs.Pointer)(0xc0005a0190), (*lfs.Pointer)(0xc000464320), (*lfs.Pointer)(0xc0004643c0), (*lfs.Pointer)(0xc0004be140), (*lfs.Pointer)(0xc0004be2d0), (*lfs.Pointer)(0xc0004be0a0), (*lfs.Pointer)(0xc0004be000), (*lfs.Pointer)(0xc0005a0000), (*lfs.Pointer)(0xc000464500), (*lfs.Pointer)(0xc0003323c0), (*lfs.Pointer)(0xc000464550), (*lfs.Pointer)(0xc0004be230), (*lfs.Pointer)(0xc000464050), (*lfs.Pointer)(0xc000332550), (*lfs.Pointer)(0xc0000ec0f0), (*lfs.Pointer)(0xc0004be1e0), (*lfs.Pointer)(0xc000464000), (*lfs.Pointer)(0xc000332370), (*lfs.Pointer)(0xc0004be280), (*lfs.Pointer)(0xc0005a0140), (*lfs.Pointer)(0xc000332690), (*lfs.Pointer)(0xc0003326e0)}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,16 @@
                                -([]*lfs.Pointer) (len=20) {
                                +([]*lfs.Pointer) (len=59) {
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "0083af118d18a63c6bb552f21d0c4ee78741f988ecd319d3cd06cb6c85a68a63",
                                +  Size: (int64) 10,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "081c39282c60731969144980d527570cdf62219e9c64dcfcbba2983eef00a80a",
                                +  Size: (int64) 30,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                  (*lfs.Pointer)({
                                @@ -10,2 +24,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "12c2deece9bdb2b80c8098728d68eb1e7bd2ee237ac26c4b5c0c945916cc1b15",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "149a2d01270f8406f1951575c6717b1e2430d529c73a014e74e6599408c27db9",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "20e108182c7ec9e6f868cd136e3b51165d2f831d7123174d4bcb2e13448eeb2e",
                                +  Size: (int64) 25,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "217b9edf66ab04650bbb438e0b8689fbea56de54a7089a24efba4ec2d0ade974",
                                @@ -17,2 +52,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "231b2ac26933b6ede876ca39be395b0e71600b632c73ad238c17bc521c5b7bd7",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "2487e4f06b0810116fdefb808bd2fc2531c837b4a85ca146642d439fecbfa1f0",
                                @@ -24,2 +66,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "2c34f06d7b5af2b406b48bc365888ac1179a176b3b95011afa5919522088f84b",
                                +  Size: (int64) 21,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "333979845b376fdf6afa0391e9156e686f9b6b2f9ca4dcf9a731e470955f1a36",
                                @@ -38,2 +87,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "373e0b9ea8863fbfea18f30b9cb50f993b5f76fe1be0cca25b1d0a49693bb911",
                                +  Size: (int64) 30,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "38aee14d2371d80cb29b10257c5b84992bdb9773a4f0493d4d5f144fddd3c9e7",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "3ac94b1d7c499747864451c13b8abc50a3a196ae54a061654a8bcc0b179327e8",
                                +  Size: (int64) 40,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "4189151928077250a9817cfbb6811c4194ff06f99d1a42fbd71ab6bda7e21a42",
                                @@ -52,2 +122,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "47181e9e688af5a47a932a3713b864c0b9caa08db164556b351f66cc61a02c79",
                                +  Size: (int64) 6,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "4752354602887adb93547d4f1c70cc74b29ba801f126663f836cf0996956fc6b",
                                @@ -59,2 +136,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "49bbca3e1981b304b6ff5ca71aea2448b505ba50853740553bf887126b0c2654",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "4b1e147ef485626268a62f22cba6fd5d741671bbb4bdaf7bf00d36dead9dfcda",
                                @@ -66,2 +150,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "51b4b333385f4080bf92a1886d0f23f2a82ea0aae0132265bf27b996e056d10c",
                                +  Size: (int64) 22,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "5793be60b7b923be93638041e91ec89d2cb4d6c087634f7ed3d4ec7041731d5a",
                                @@ -73,2 +164,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "599712a0858056905f65f47379a656ac40e2429cb87d6ca653986632aa208059",
                                +  Size: (int64) 32,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "5def0a8e313247dd8c9cf49f1c3eacdeaf86199382494a162459f5aaa396b3c7",
                                +  Size: (int64) 40,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "61f2a9e519d731891da52aab4a84dcc044af59b3b165c9287d397df0e8012007",
                                +  Size: (int64) 42,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "647fc88fa01959a5b455eb4e21a3b3ec729cd467c51198c5144cbf55554bfefa",
                                @@ -80,2 +192,58 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "65f577efd0e59f25a1ee0a3229261b189d4e14d461a324c671dd3182525a4f35",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "6c79525a3249c63a06496e266c8b8212cc1f0e260bcf9bd4cad6dc4a71d1bd34",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "735ad4fbf949394106c091a78e47fd25d46f03ef29e17d6e7879d8e5a54d6a76",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "73701626ab91b268bb6830d8d1ca414b0e476786ecb4e2f3f9693c3e4f4ab605",
                                +  Size: (int64) 30,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "826efb18ae20e99a3916053d483629d4a8db495d0e2af8d00cd66f90be588c24",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "85cce12a9ef7f0dd0f31cf966101ed3049a17d664fb4ba449f31c0513514fe65",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "899b2b5dc70763f362aeb09374d654a937707b930ad21d3531ab6f29181dd508",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "91c7661e0ab6bcfe6bffa0fa02998b3467d8814e43ac0558adb3ebf86b490b3c",
                                +  Size: (int64) 32,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "9668c5a232c8f11ba601af1bf1b4bdc28795bd53c1113138ca75d8ec850f18f2",
                                @@ -87,2 +255,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "974d50fe68c7a0b9958e1ad5247a59b3d311c39adabd1781b53adb9d629dc200",
                                +  Size: (int64) 25,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "9b9906f5ae3adb383e5353dd2fa814bf10c2fdc3a8e607e77e0436ec45573ea8",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "9d82a50c56f74a863163213fb2ddd980946ead26d03f1d68500ce1ea724e7c01",
                                +  Size: (int64) 31,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "9e366e58df4301d5ceeb4ebee8fd6d4402305ad7b9d6b549de40f3b4de8cf3a6",
                                @@ -94,2 +283,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "b4faa9f0b604b349783682706a279715988c2f6447918290f46ebd6e990b2b46",
                                +  Size: (int64) 25,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "bd6174d17b30f8dd7e0f19f0553091d33267773fa85bef7b71d8a64d0cb20efd",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "bff3b771ec7a13b34533022e84895d3c45e1b99157c14a935a5054089c9857e1",
                                +  Size: (int64) 40,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "c3990ba4bc01a7c0e1d659f8ad4bc1886383246cf85514dfa39508dbce6f6bd3",
                                @@ -101,2 +311,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "c3fb7da348eabe72c12eba33319de9fb80fa97a77788b7da3df1545894351e28",
                                +  Size: (int64) 22,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "cccd4b047ed4a6fa5619edfa66d827862e9d637651e28c9aa78131752fda4f3e",
                                @@ -108,2 +325,23 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "d7e099738e67f285241298b4ded39e6db48cc2008fd1960215f3c0403eae3427",
                                +  Size: (int64) 12,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "dcd40b25a27daef8ee0a92aef055376e72c20e227d383b0e124a79b76be13af7",
                                +  Size: (int64) 16,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "e174da7488e577a5e1fed5bc083f07c927df5e061e5a4c50254898abf1c3c81c",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "e9d38a101d120a86571af61375b92bc5d219d4e74d205031a021867b0c8ecdb7",
                                @@ -122,2 +360,9 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "ef87221779ec2e218b3a6ef4e1dec9f743b71530b2e2fc561ccdb469195ee800",
                                +  Size: (int64) 25,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "f2c2add61f326fa2b1c0420daa5664d423bf82d0fe517b6700879801be0e6b42",
                                @@ -129,4 +374,32 @@
                                   Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "f3c853135aa21c3481c82e917208f0fae5f91eda38032e01dbff95b9c7404c17",
                                +  Size: (int64) 25,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                   Oid: (string) (len=64) "f57e428d80e5b6b6359b3d9b5b111e4f31af6f3643a6ab2621a0d3301a3e728e",
                                   Size: (int64) 30,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "f781253591c98cdca2e3069f9414579f900205ea3ee435841a0fc6a12c5f165c",
                                +  Size: (int64) 20,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "f859b587469627f138d534b20db381d9f415e0eb138c90914e0f2da33bb2d870",
                                +  Size: (int64) 36,
                                +  OidType: (string) (len=6) "sha256",
                                +  Extensions: ([]*lfs.PointerExtension) <nil>
                                + }),
                                + (*lfs.Pointer)({
                                +  Version: (string) (len=34) "https://git-lfs.github.com/spec/v1",
                                +  Oid: (string) (len=64) "f96fdeff073a3145d72b481d532091e3149f0cbea96ce5f8defc2cdce42f46c2",
                                +  Size: (int64) 32,
                                   OidType: (string) (len=6) "sha256",
                Test:           TestAllCurrentObjectsSome
                Messages:       Oids from disk should be the same as in commits
FAIL
FAIL    github.com/git-lfs/git-lfs/lfs  0.287s
```

**System environment**

I believe this bug can be reproduced in any OS.

--- a/Makefile
+++ b/Makefile
@@ -409,8 +409,16 @@ test-race : GO_TEST_EXTRA_ARGS=-race
 #
 # And so on.
 test : fmt $(.DEFAULT_GOAL)
-	(unset GIT_DIR; unset GIT_WORK_TREE; \
-	$(GO) test -count=1 $(GO_TEST_EXTRA_ARGS) $(addprefix ./,$(PKGS)))
+	( \
+		unset GIT_DIR; unset GIT_WORK_TREE; unset XDG_CONFIG_HOME; \
+		tempdir="$$(mktemp -d)"; \
+		export HOME="$$tempdir"; \
+		export GIT_CONFIG_NOSYSTEM=1; \
+		$(GO) test -count=1 $(GO_TEST_EXTRA_ARGS) $(addprefix ./,$(PKGS)); \
+		RET=$$?; \
+		rm -fr "$$tempdir"; \
+		exit $$RET; \
+	)
 
 # integration is a shorthand for running 'make' in the 't' directory.
 .PHONY : integration

--- a/t/testenv.sh
+++ b/t/testenv.sh
@@ -128,12 +128,14 @@ GIT_CONFIG_NOSYSTEM=1
 GIT_TERMINAL_PROMPT=0
 GIT_SSH=lfs-ssh-echo
 APPVEYOR_REPO_COMMIT_MESSAGE="test: env test should look for GIT_SSH too"
+LC_ALL=C
 
 export CREDSDIR
 export GIT_LFS_FORCE_PROGRESS
 export GIT_CONFIG_NOSYSTEM
 export GIT_SSH
 export APPVEYOR_REPO_COMMIT_MESSAGE
+export LC_ALL
 
 # Don't fail if run under git rebase -x.
 unset GIT_DIR

